### DISCUSSION TITLE
Only 1 CPU is used when training, batch size 20, num_workers 20

### DISCUSSION THREAD

**Discussion by davodogster** [pyg-team/pytorch_geometric#4840] [Q&A]
Is this normal? It seems like I am wasting CPU resources

**Comment by rusty1s** [comment]
The `DataLoader` should use 20 workers, and the GNN should use multi-threading for heavy computations as well (except for `scatter_*` ops which are single-threaded). If you are only using 1 CPU with `num_workers=20`, I assume there is something wrong with your PyTorch installation?

**Reply by davodogster** [reply]
Do you thinks it's because I'm training with InMemory?
![image](https://user-images.githubusercontent.com/46079516/175179779-33d8f3d0-7bf9-46f2-a08a-174501e3c9ab.png)

**Reply by rusty1s** [reply]
This shouldn't be related AFAIK.
